router/xgress/xgress.go

```diff
@@ -580,16 +581,15 @@ func (self *Xgress) SendEmptyAck() {
 	acker.ack(ack, self.address)
 }
 
-func (self *Xgress) Inspect() map[string]interface{} {
-	result := map[string]interface{}{}
-	result["type"] = "xgress"
-	result["addr"] = string(self.address)
-	result["originator"] = self.originator.String()
-
+func (self *Xgress) InspectCircuit(detail *inspect.CircuitInspectDetail) {
 	timeSinceLastRxFromLink := time.Duration(info.NowInMilliseconds()-atomic.LoadInt64(&self.timeOfLastRxFromLink)) * time.Millisecond
-	result["timeSinceLastLinkRx"] = timeSinceLastRxFromLink.String()
-	result["sendBuffer"] = self.payloadBuffer.Inspect()
-	result["recvBuffer"] = self.linkRxBuffer.Inspect()
+	xgressDetail := &inspect.XgressDetail{
+		Address:             string(self.address),
+		Originator:          self.originator.String(),
+		TimeSinceLastLinkRx: timeSinceLastRxFromLink.String(),
+		SendBufferDetail:    *self.payloadBuffer.Inspect(),
```

Comment:
any reason to not have detail fields pointers?

Reply:
I liked guaranteeing they wouldn't be null, but they're always going to be non-nil just based on usage, so probably makes sense to just use pointers. Updated.